### ISSUE TITLE
System.InvalidOperationException has been thrown 

### ISSUE THREAD

**Issue by devology-rob** [upta/pubsub#30]
Hi,

I successfully upgraded from PubSub 3.1.0 to 4.0.0, but this week I un-installed the app and re-installed from scratch.  During startup I received

```
System.InvalidOperationException has been thrown
Collection was modified; enumeration operation may not execute
```

Oddly it seemed to crash every time thereafter.

I tried putting in lock mutex's around all of my hub.Subscribe and hub.Publish invocations to ensure no threading issues from my end, but the problem continued.

In the end I had to revert back to PubSub 3.1.0 and then the problem went away.  

Unfortunately I don't have the call stack any more, but it was deep inside PubSub with a LINQ .Where clause.

<img width="1006" alt="Screenshot 2019-12-15 at 17 08 08" src="https://user-images.githubusercontent.com/18516600/70868540-abad8100-1f79-11ea-8b09-b7b4da89a1d9.png">

**Comment by christophedelanghe** [comment]
What is the status here? I have the same problem ... is there a workaround?

**Comment by gb0o** [comment]
@upta Can you merge pull request https://github.com/upta/pubsub/pull/31 to fix this issue?

**Comment by upta** [comment]
@gb0o @christophedelanghe @devology-rob 

My apologies!  I completely forgot this fix was pending, then was going to do it when @christophedelanghe brought it up and got busy and forgot again :(

I'll merge the fix and update the nuget package tomorrow.  Sorry about the inconvenience!

**Comment by christophedelanghe** [comment]
Thx! I reverted back to version 3.1.0. Wil test it again after your merge.

**Comment by upta** [comment]
Merged in #31 and published to Nuget as 4.0.1

Thanks guys!
